go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -529,6 +508,31 @@ func (bsnc *BaseSecondaryNetworkController) removePodForSecondaryNetwork(pod *ka
 	return nil
 }
 
+// hasIPAMClaim determines whether a pod's IPAM is being handled by IPAMClaim CR.
+// pod passed should already be validated as having a network connection to nadName
+func (bsnc *BaseSecondaryNetworkController) hasIPAMClaim(pod *kapi.Pod) (bool, error) {
+	if !bsnc.IsPrimaryNetwork() || !bsnc.AllowsPersistentIPs() {
```

Comment:
we don't need to do this for secondary networks? i thought we support ipam claim for secondary as well? - but it certainly seems like we were doing the earlier logic only for active networks of the pod...

Reply:
I'm also confused here .. 

> we don't need to do this for secondary networks? i thought we support ipam claim for secondary as well? - but it certainly seems like we were doing pthe logic only for active networks of the pod in the older code as well..

For secondary networks, the `IPAMClaim` will be pointed by the `IPAMClaimReference` [attribute](https://github.com/k8snetworkplumbingwg/network-attachment-definition-client/blob/7d2def1df606637b28a3d8cde6793025f3bd053e/pkg/apis/k8s.cni.cncf.io/v1/types.go#L170) defined in the network selection element, as indicated in the multi-network de-facto standard.

For sure this code is required *also* for secondary networks.

Reply:
So in the current code:
```
hasIPAMClaim := network != nil && network.IPAMClaimReference != ""
```

network.IPAMClaimReference is set in 
```
_, networkMap, err := util.GetPodNADToNetworkMappingWithActiveNetwork(pod, bsnc.NetInfo, activeNetwork)
```

In there we get GetPodNADToNetworkMapping, which returns the network selection elements for the pod, keyed by NAD, then if the active network is not default, we add that network to the map and if the pod has the annotation:

```
OvnUDNIPAMClaimName = "k8s.ovn.org/primary-udn-ipamclaim"
```

we add its ipamclaim reference to the network.

I missed that the secondary ipam claim reference was stored in the network selection element and thought you were looking for primary only. I will modify my code to account for this!

Reply:
@trozet let me know once you have re-worked it.